onfido/utils.py

```diff
@@ -1,19 +1,21 @@
 
-def flat_nested_dict(data):
-    """ Flat nested dictionary """
+def form_data_converter(data):
+    """ Flat nested dictionary and stringify booleans for form-data """
 
-    # Check if any value is a dictionary
-    if any(type(item) is dict for item in data.values()):
+    # If no dict or booleans among values, no conversion needed
+    if not any(type(item) in (dict, bool) for item in data.values()):
+        return data
 
-        # Start copying not nested values
-        result = {k: v for k, v in data.items() if type(v) is not dict}
+    result = {}
 
-        # And add nested ones with proper format
-        for root_k, root_v in data.items():
-            if type(root_v) is dict:
-                for k, v in root_v.items():
-                    result['{}[{}]'.format(root_k, k)] = v
+    # Perform needed conversion for each
+    for k, v in data.items():
+        if type(v) is dict:
+            for child_k, child_v in v.items():
+                result['{}[{}]'.format(k, child_k)] = child_v
```

Comment:
It would not handle further nested structure but it would probably good enough as there isn't any of them in involved APIs (documents and live_photos upload).

Reply:
There indeed are not any right _now_, but given that it’s simple to fix it for both cases then we might as well?

Reply:
Sounds good!